### PR TITLE
DX: add types to anonymous functions

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

**Review by Wirone — APPROVED** [review]

File: src/Fixer/ClassNotation/OrderedTypesFixer.php [hunk 1]
```diff
@@ -308,7 +308,7 @@ private function isTypeSortable(TypeAnalysis $type): bool
      */
     private function collectDisjunctiveNormalFormTypes(string $type): array
     {
-        $types = array_map(static function ($subType) {
```
Comment (keradus) [inline src/Fixer/ClassNotation/OrderedTypesFixer.php:311]: any SCA to prevent adding another not-typed lambdas?

Comment (kubawerlos) [reply src/Fixer/ClassNotation/OrderedTypesFixer.php:311]: I've spotted these with [slam/phpstan-extensions](https://github.com/Slamdunk/phpstan-extensions) but was not able to fix all of the types (some lambdas need multiple types and with PHP 7.4 we cannot have them).

Comment (keradus) [reply src/Fixer/ClassNotation/OrderedTypesFixer.php:311]: @kubawerlos 
maybe for those cases we can add `mixed` typehint, and prevent adding any more untyped lambdas?
then, we deal with `mixed` afterwars

Comment (kubawerlos) [reply src/Fixer/ClassNotation/OrderedTypesFixer.php:311]: Definitely, as soon as we drop PHP 7.4 support.

**Pull request merged**